### PR TITLE
[Compute] Adding sharing profile to New-AzCapacityReservationGroup and Update-AzCapacityReservationGroup

### PR DESCRIPTION
**Pull request by haagha**
## Description

Resolves https://github.com/Azure/azure-powershell-cmdlet-review-pr/issues/1390

## Mandatory Checklist

- Please choose the target release of Azure PowerShell
  - [x] [General release](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Public preview](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Private preview](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Engineering build](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] N/A

- [ ] Check this box to confirm: **I have read the [_Submitting Changes_](../blob/main/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/main/CONTRIBUTING.md) and reviewed the following information:**

* **SHOULD** update `ChangeLog.md` file(s) appropriately
    * For SDK-based development mode, update `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`.
        * A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header in the past tense. 
    * For autorest-based development mode, include the changelog in the PR description.
    * Should **not** change `ChangeLog.md` if no new release is required, such as fixing test case only.
* **SHOULD** regenerate markdown help files if there is cmdlet API change. [Instruction](../blob/main/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)
* **SHOULD** have proper test coverage for changes in pull request.
* **SHOULD NOT** adjust version of module manually in pull request

### CODE REVIEW

**Comment by vidai-msft** [comment]
@haagha When this PR is ready for us to review, Please assign it to me. Thanks!

File: src/Compute/Compute.Test/ScenarioTests/VirtualMachineTests.ps1 [hunk 1]
```diff
@@ -7441,3 +7441,50 @@ function Test-VMTLWithGallerySourceImage
         Clean-ResourceGroup $rgname;
     }
 }
+
+
+<#
+.SYNOPSIS
+Windows machine test ensuring the EnableAutoUpdate value on the
```
Comment (Sandido) [inline src/Compute/Compute.Test/ScenarioTests/VirtualMachineTests.ps1:7448]: Update the description so we understand what this test is for.

File: src/Compute/Compute.Test/ScenarioTests/VirtualMachineTests.ps1 [hunk 2]
```diff
@@ -7441,3 +7441,50 @@ function Test-VMTLWithGallerySourceImage
...
+<#
+.SYNOPSIS
+Windows machine test ensuring the EnableAutoUpdate value on the 
+provided VM is not overwritten. 
+#>
+function Test-CapacityReservationSharingProfile
+{
+    # Setup
+    $rgname = Get-ComputeTestResourceName;
+    $loc = 'WestEurope';
```
Comment (Sandido) [inline src/Compute/Compute.Test/ScenarioTests/VirtualMachineTests.ps1:7455]: use the method that gets a location automatically. Any reason to hardcode WestEurope?

File: src/Compute/Compute/Generated/CapacityReservation/NewAzCapacityReservationGroupCommand.cs [hunk 3]
```diff
@@ -70,6 +70,10 @@ public class NewAzureCapacityReservationGroup : ComputeAutomationBaseCmdlet
             Mandatory = false)]
         public string[] Zone { get; set; }
 
+        [Parameter(
+            Mandatory = false)]
```
Comment (Sandido) [inline src/Compute/Compute/Generated/CapacityReservation/NewAzCapacityReservationGroupCommand.cs:74]: Add in ValueFromPipelineByPropertyName  and HelpMessage. The helpMessage should match the description on the help file.

File: src/Compute/Compute/help/New-AzCapacityReservationGroup.md [hunk 4]
```diff
@@ -92,6 +92,21 @@ Accept pipeline input: True (ByPropertyName)
 Accept wildcard characters: False
 ```
 
+### -ProgressAction
+{{ Fill ProgressAction Description }}
```
Comment (Sandido) [inline src/Compute/Compute/help/New-AzCapacityReservationGroup.md:96]: Why is this here?

File: src/Compute/Compute/help/Update-AzCapacityReservationGroup.md [hunk 5]
```diff
@@ -14,28 +14,29 @@ Update the Capacity Reservation Group.
...
+ [<CommonParameters>]
 ```
 
 ### DefaultParameterSet
 ```
-Update-AzCapacityReservationGroup -ResourceGroupName <String> -Name <String> [-AsJob] [-Tag <Hashtable>]
- [-DefaultProfile <IAzureContextContainer>] [-WhatIf] [-Confirm]
- [<CommonParameters>]
+Update-AzCapacityReservationGroup -ResourceGroupName <String> -Name <String> [-SharingProfile <String[]>]
+ [-AsJob] [-Tag <Hashtable>] [-DefaultProfile <IAzureContextContainer>] [-ProgressAction <ActionPreference>]
```
Comment (Sandido) [inline src/Compute/Compute/help/Update-AzCapacityReservationGroup.md:25]: WHy is ProgressAction here?

**Comment by grizzlytheodore** [comment]
other than adam's comments, looks good to me, if the testing is thorough with use of a list as the parameter input

File: src/Compute/Compute/help/New-AzCapacityReservationGroup.md (hunk 4 above)
Comment (vidai-msft) [reply src/Compute/Compute/help/New-AzCapacityReservationGroup.md:96]: @Sandido This is a bug introduced by platyPS. We will take care of it.

File: src/Compute/Compute/help/Update-AzCapacityReservationGroup.md (hunk 5 above)
Comment (vidai-msft) [reply src/Compute/Compute/help/Update-AzCapacityReservationGroup.md:25]: Same as above. We will remove it after the PR is merged.

**Comment by vidai-msft** [comment]
@haagha Please address the review comments.

File: src/Compute/Compute.Test/ScenarioTests/VirtualMachineTests.ps1 (hunk 2 above)
Comment (haagha) [reply src/Compute/Compute.Test/ScenarioTests/VirtualMachineTests.ps1:7455]: feature is available in limited locations at the moment behind an AFEC flag

File: src/Compute/Compute/Generated/CapacityReservation/NewAzCapacityReservationGroupCommand.cs (hunk 3 above)
Comment (haagha) [reply src/Compute/Compute/Generated/CapacityReservation/NewAzCapacityReservationGroupCommand.cs]: ValueFromPipelineByPropertyName default is false, added help.

File: src/Compute/Compute.Test/ScenarioTests/VirtualMachineTests.ps1 (hunk 1 above)
Comment (haagha) [reply src/Compute/Compute.Test/ScenarioTests/VirtualMachineTests.ps1]: done

**Comment by haagha** [comment]
> @haagha Please address the review comments.

@vidai-msft comments have been addressed, please merge. thanks!

**Review by vidai-msft — APPROVED** [review]
LGTM

**Review by Sandido — APPROVED** [review]
